### ISSUE TITLE
How to enable server certificate verification on the Nodejs clients like MetricServiceClient/KeyManagementServiceClient for Mutual TLS ? 

### ISSUE THREAD

**Issue by Ruthwik** [googleapis/google-cloud-node#5453]
We are using Nodejs MetricServiceClient/KeyManagementServiceClient client in our product. How do we enable server certificate verification in the MetricServiceClient/KeyManagementServiceClient client?
We want to enable client-side verification of server certificates with our CA bundle.
We've explored the options of MetricServiceClient/KeyManagementServiceClient but couldn't find any.

In the case of other cloud providers we have found a way to pass the CA bundle in the options where the client verifies.

Example for AWS

```
const nodeOptions = {
    httpsAgent: new https.Agent({
        rejectUnauthorized: true,
        ca: caBundle,
        maxVersion
    }),
};

const requestHandler =  new NodeHttpHandler(nodeOptions);
const athena = new AthenaClient({ credentials, region : 'us-east-1', requestHandler });
```

In the above example, If rejectUnauthorized is true the server will reject any connection which is not authorized with the list of supplied CAs. I also found the it is different for services and client in google nodejs SDK.

@beccasaurus @blowmage

**Comment by guru1306** [comment]
I am also facing the similar issue. Did you find any solution?

**Comment by sofisl** [comment]
I don't think this can be done through grpc, [see](https://cloud.google.com/load-balancing/docs/ssl-certificates/self-managed-certs#use_a_self-signed_certificate). However, I do think this could be done using our REST transport. You should be able to configure a client to use [mTLS](https://github.com/googleapis/gaxios/blob/57077e0255017d719862991c71582241690a7b60/src/common.ts#L217) or use the [agent parameter](https://github.com/googleapis/gaxios/blob/57077e0255017d719862991c71582241690a7b60/src/common.ts#L202) like bigquery. In order to do this you'll need to instantiate using the googleapis library, like so:

```
const {google} = require('googleapis/cloudkms');
const cloudkms = google.cloudkms('v1');
```

and in the request, you can send a second parameter (options) that extends [GaxiosOptions](https://github.com/googleapis/google-api-nodejs-client/blob/957bc850439a639c5534957333b09598944952c3/src/apis/cloudkms/v1.ts#L4937).

**Comment by sofisl** [comment]
@Ruthwik did this work for you?

**Comment by Ruthwik** [comment]
@sofisl I will post here once I try these changes. It requires a change of our current libraries.

**Comment by sofisl** [comment]
Yes, it should. Closing for now to try and keep queue clean - please open a new issue if this doesn't work for you!
